hawc/tools/tables/generic.py

```diff
@@ -0,0 +1,80 @@
+from enum import IntEnum
+from typing import List
+
+from docx import Document
+from pydantic import BaseModel, conint, validator
+
+from .parser import QuillParser
+
+
+class CellEnum(IntEnum):
+    CELL = 1
+    EMPTY = 2
+    SKIP = 3
+
+
+class GenericTableCell(BaseModel):
+    header: bool = False
+    row: conint(ge=0)
+    column: conint(ge=0)
+    row_span: conint(ge=1) = 1
+    col_span: conint(ge=1) = 1
+    html_text: str
+
+    def __str__(self):
+        return "CELL STRING"
+
+    def row_order_index(self, rows):
+        return (self.column) * rows + self.row
+
+    def to_docx(self, block):
+        parser = QuillParser()
+        return parser.feed(self.html_text, block)
+
+
+class GenericTable(BaseModel):
+    rows: conint(gt=0)
+    columns: conint(gt=0)
+    cells: List[GenericTableCell]
+
+    @validator("cells")
+    def validate_cells(cls, cells, values):
```

Comment:
i noticed sometimes that this would run if i didnt specify columns, might add `skip_on_failure=True` to see if that fixes, and maybe a few unit-tests to show good/bad data examples?